src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -597,6 +644,87 @@ def iot_hub_consumer_group_delete(client, hub_name, consumer_group_name, resourc
     return client.iot_hub_resource.delete_event_hub_consumer_group(resource_group_name, hub_name, event_hub_name, consumer_group_name)
 
 
+def iot_hub_identity_assign(cmd, client, hub_name, identities, identity_role=None, identity_scopes=None, resource_group_name=None):
+    resource_group_name = _ensure_resource_group_name(client, resource_group_name, hub_name)
+
+    def getter():
+        return iot_hub_get(cmd, client, hub_name, resource_group_name)
+
+    def setter(hub):
+        user_identities = [i for i in identities if i != SYSTEM_IDENTITY]
+
+        if user_identities and not hub.identity.user_assigned_identities:
+            hub.identity.user_assigned_identities = {}
+        for identity in user_identities:
+            hub.identity.user_assigned_identities[identity] = hub.identity.user_assigned_identities.get(identity, {}) if hub.identity.user_assigned_identities else {}
+
+        has_system_identity = hub.identity.type in [IdentityType.system_assigned_user_assigned.value, IdentityType.system_assigned.value]
+
+        if SYSTEM_IDENTITY in identities or has_system_identity:
+            hub.identity.type = IdentityType.system_assigned_user_assigned.value if hub.identity.user_assigned_identities else IdentityType.system_assigned.value
+        else:
+            hub.identity.type = IdentityType.user_assigned.value if hub.identity.user_assigned_identities else IdentityType.none.value
+
+        poller = client.iot_hub_resource.begin_create_or_update(resource_group_name, hub_name, hub, {'IF-MATCH': hub.etag})
+        return LongRunningOperation(cmd.cli_ctx)(poller)
+
+    if bool(identity_role) ^ bool(identity_scopes):
+        raise CLIError('At least one scope (--scopes) and one role (--role) required for system-managed identity role assignment')
```

Comment:
Please use the specific error type `RequiredArgumentMissingError` instead of `CLIError`.